gtsfm/frontend/correspondence_generator/mast3r_correspondence_generator.py

```diff
@@ -0,0 +1,341 @@
+"""Correspondence generator using the MASt3R model.
+
+https://github.com/naver/mast3r
+
+Authors: Akshay Krishnan
+"""
+
+from typing import Any, Dict, List, Tuple
+from pathlib import Path
+import sys
+
+HERE_PATH = Path(__file__).parent
+MAST3R_REPO_PATH = HERE_PATH.parent.parent.parent / "thirdparty" / "mast3r"
+if MAST3R_REPO_PATH.exists():
+    # workaround for sibling import
+    sys.path.insert(0, str(MAST3R_REPO_PATH))
+else:
+    raise ImportError(
+        f"mast3r is not initialized, could not find: {MAST3R_REPO_PATH}.\n "
+        "Did you forget to run 'git submodule update --init --recursive' ?"
+    )
+
+from mast3r.model import AsymmetricMASt3R
+import mast3r.cloud_opt.sparse_ga as mast3r_ga
+
+import numpy as np
+import torch
+import torchvision.transforms as tvf
+from dask.distributed import Client, Future
+
+from gtsfm.common.image import Image
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.utils import images as image_utils
+from gtsfm.utils import logger as logger_utils
+from gtsfm.frontend.correspondence_generator.correspondence_generator_base import (
+    CorrespondenceGeneratorBase,
+)
+
+
+logger = logger_utils.get_logger()
+_MODEL_PATH = str(
+    Path(__file__).resolve().parent.parent.parent.parent
+    / "thirdparty"
+    / "mast3r"
+    / "checkpoints"
+    / "MASt3R_ViTLarge_BaseDecoder_512_catmlpdpt_metric.pth"
+)
+GRID_DIVISOR = 16
+PATCH_MULTIPLIER = 8
+
+def resize_image(img: Image, long_edge_size: int) -> Image:
+    """Resizes image such that longest edge is equal to long_edge_size.
+
+    Args:
+        img: The input image to be resized.
+        long_edge_size: The desired size for the longest edge of the image.
+
+    Returns:
+        The resized image.
+    """
+    max_size = max(img.height, img.width)
+    ratio = float(long_edge_size) / max_size
+    new_height = int(img.height * ratio)
+    new_width = int(img.width * ratio)
+    new_image = image_utils.resize_image(img, new_height, new_width)
+    return new_image
```

Comment:
Do we not have a function for this already? If not, maybe we should move this to image utils

Reply:
I dont think we have a function for resizing acc to longest edge. Sure, moved it.